src/app/header/header.component.ts

```diff
@@ -36,7 +38,9 @@ export class HeaderComponent implements OnInit {
     public context: ContextService
   ) {
     router.events.subscribe((val) => {
-      this.onNavigate();
+      if (val instanceof NavigationEnd) {
```

Comment:
That's a type assertion I think, wheras I need a type guard (see https://www.typescriptlang.org/docs/handbook/advanced-types.html)